### ISSUE TITLE
Relay didn't reconnect after a service/internet outage

### ISSUE THREAD

**Issue by jedireza** [spacekit/spacekit#50]
We should test this more. The goal should be that relays reconnect within a minute of the service coming back online or the internet connection being restored.

**Comment by mcav** [comment]
Was it trying to backoff, or did it not recognize that the connection was terminated?

**Comment by jedireza** [comment]
I'm not sure yet. It was my relay at work and I'm not at the office.

**Comment by mcav** [comment]
```
{"name":"SpaceKit","hostname":"mcav.local","pid":19499,"module":"Relay","level":50,"err":{"message":"connect ECONNREFUSED 52.21.84.241:443","name":"Error","stack":"Error: connect ECONNREFUSED 52.21.84.241:443\n    at Object.exports._errnoException (util.js:953:11)\n    at exports._exceptionWithHostPort (util.js:976:20)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1080:14)","code":"ECONNREFUSED"},"msg":"Service connection encountered an error.","time":"2016-05-25T17:28:40.232Z","v":0}

/Users/marcus/moz/spacekit/node_modules/precond/lib/checks.js:15
    throw error;
    ^
IllegalStateError: Backoff in progress.
    at Backoff.backoff (/Users/marcus/moz/spacekit/node_modules/backoff/lib/backoff.js:38:13)
    at Relay.maybeReconnect (/Users/marcus/moz/spacekit/lib/index.js:57:20)
    at WebSocket.ws.on (/Users/marcus/moz/spacekit/lib/index.js:109:12)
    at emitTwo (events.js:106:13)
    at WebSocket.emit (events.js:191:7)
    at WebSocket.cleanupWebsocketResources (/Users/marcus/moz/spacekit/node_modules/ws/lib/WebSocket.js:950:8)
    at ClientRequest.onerror (/Users/marcus/moz/spacekit/node_modules/ws/lib/WebSocket.js:712:31)
    at emitOne (events.js:96:13)
    at ClientRequest.emit (events.js:188:7)
    at TLSSocket.socketErrorListener (_http_client.js:306:9)
    at emitOne (events.js:96:13)
    at TLSSocket.emit (events.js:188:7)
    at emitErrorNT (net.js:1272:8)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

**Comment by mcav** [comment]
Fixed in #53
